### PR TITLE
Handle composite keys

### PR DESCRIPTION
**Pull request by jamsilver**
Trying to run scrubber against a Drupal site and all of the field tables use 4 columns as a (composite) primary key. This Merge Request adds support for this feature.

This is an API breaking change as it changes the DatabaseUpdate interface to accept a DatabaseKey object instead of a string|int for $primaryKey.

### CODE REVIEW

**Comment by jamsilver** [comment]
Trying to use this against a Drupal site and all of the field tables use 4 columns as a (composite) primary key. This tries to add support for this feature.

This is an API breaking change as it changes the DatabaseUpdate interface to accept a DatabaseKey object instead of a string|int for $primaryKey.

File: src/Handlers/Handler.php [hunk 1]
```diff
@@ -24,14 +26,14 @@ public function __construct(
         string $table,
         string $field,
         mixed $input,
-        string $primaryKey = 'id',
+        string $seed = 'scrubber',
+        ?DatabaseKey $primaryKey = null,
         ?string $type = null,
-        string $seed = 'scrubber'
```
Comment (jamsilver) [inline src/Handlers/Handler.php:29]: The order of parameters here didn't seem to match the DataHandler interface, so I've switched them around to match.

File: tests/Support/Fakes/Database.php [hunk 2]
```diff
@@ -13,37 +14,33 @@ public function __construct()
         $this->entries = [];
     }
 
-    public function fetch(string $table, string $primaryKey): array //phpcs:ignore
+    public function fetch(string $table, DatabaseKey $primaryKey): array //phpcs:ignore
     {
+        print 'ARG: ' . $table;
```
Comment (jamsilver) [inline tests/Support/Fakes/Database.php]: Oops, need to remove this.

File: tests/Unit/ScrubberTest.php [hunk 3]
```diff
@@ -44,119 +46,133 @@ public function it_can_run_the_scrubber_using_the_test_config_and_get_the_expect
 
         $scrubber->run();
 
-        $this->assertCount(16, $this->database->entries);
+        $this->assertCount(18, $this->database->entries);
         $this->assertEquals([
             [
                 'table' => 'users',
                 'field' => 'first_name',
-                'value' => 'Keyshawn',
```
Comment (jamsilver) [inline tests/Unit/ScrubberTest.php:52]: I don't know why, but my test faked different data here. I just changed the values to match. I wonder why this happened?

File: composer.lock [hunk 4]
```diff
@@ -0,0 +1,3251 @@
+{
```
Comment (jamsilver) [inline composer.lock]: Ah, I guess I should not have committed the composer.lock.

File: src/Handlers/Handler.php (hunk 1 above)
Comment (clnt) [reply src/Handlers/Handler.php:29]: Good spot, I have moved the `seed` parameter back to the end of the call as I feel it's less related to the database and is just for faker data. Plus I think having the `primaryKey` as the first optional parameter makes more sense.

File: tests/Support/Fakes/Database.php (hunk 2 above)
Comment (clnt) [reply tests/Support/Fakes/Database.php]: Removed

File: tests/Unit/ScrubberTest.php (hunk 3 above)
Comment (clnt) [reply tests/Unit/ScrubberTest.php:52]: The values usually change when another item is added to the data as the values faker returns are based off an iterator and the seed string which is set.

File: composer.lock (hunk 4 above)
Comment (clnt) [reply composer.lock]: I think I have been able to remove this from history and add a .gitignore file in place, no worries

**Pull request merged**